cmd/networkaccessmanager.go

```diff
@@ -0,0 +1,327 @@
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func getNetworkByID(networkID string) error {
+	api := networkaccessmanager.New(curl())
+
+	result, err := api.GetNetworkTargetByID(networkID)
+	if err != nil {
+		return err
+	}
+
+	return stdout(result)
+}
+
+//
+//
+//
+func updateNetworkCmd() *cobra.Command {
+	var networkID string
+
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update a specific netword target",
+		Long:  `Update a specific netword target`,
+		Example: `
+	privx-cli nam update JSON-FILE --id <NETWORK-ID>
+		`,
+		Args:         cobra.ExactArgs(1),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return updateNetwork(args, networkID)
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVar(&networkID, "id", "", "id of the target network")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func updateNetwork(args []string, targetID string) error {
+	var network *networkaccessmanager.Item
+
+	api := networkaccessmanager.New(curl())
+
+	err := decodeJSONnam(args[0], &network)
+	if err != nil {
+		return err
+	}
+
+	err = api.UpdateNetworkTarget(network, targetID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//
+//
+//
+func deleteNetworkByIDCmd() *cobra.Command {
+	var networkID string
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete a network by ID",
+		Long:  `Delete a network by ID`,
+		Example: `
+	privx-cli nam delete --id <NETWORK-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deleteNetworkByID(networkID)
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVar(&networkID, "id", "", "id of the target network to delete")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func deleteNetworkByID(networkID string) error {
+	api := networkaccessmanager.New(curl())
+
+	err := api.DeleteNetworkTargetByID(networkID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//
+//
+//
+func disableNetworkByIDCmd() *cobra.Command {
+	var networkID string
+	var disableState bool
+	cmd := &cobra.Command{
+		Use:   "disable",
+		Short: "disable or enable a network by ID",
+		Long:  `disable or enable a network by ID`,
+		Example: `
+	privx-cli nam delete --id <NETWORK-ID> --disable <true or false>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return disableNetworkByID(disableState, networkID)
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVar(&networkID, "id", "", "id of the target network to delete")
+	flags.BoolVar(&disableState, "disable", true, "disable true or false")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func disableNetworkByID(disableState bool, networkID string) error {
+	api := networkaccessmanager.New(curl())
+
+	err := api.DisableNetworkTargetByID(disableState, networkID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//
+//
+//
+func decodeJSONnam(name string, object interface{}) error {
```

Comment:
Okay I understand that there are different ways but which one should I use? I don't mind using any of them but i don't feel  like i should be the one to take this decision.
Also should we add that function to `cmd/root.go` ? (I suggested that since there are functions there like `stdout() `used in the whole code.

Reply:
If you do not mind, move this function into a separate file "utils.go" under "cmd" folder and make it within the "cmd" package. stdout function should be there, too but it is not within the current scope.